src/compiler/checker.ts

```diff
@@ -7074,9 +7079,6 @@ namespace ts {
                 if (t.flags & TypeFlags.Substitution) {
                     return getBaseConstraint((<SubstitutionType>t).substitute);
                 }
-                if (isGenericMappedType(t)) {

```

Comment:
This makes the constraint of a generic mapped type itself, right? I guess that makes sense since indexed accesses work that way already.

Reply:
Correct. I'm not sure why this code was here, but it definitely isn't right. Removing it had no effect on baselines (including RWC baselines), but it fixes the second issue I found while exploring #26409.